### ISSUE TITLE
Forge only version

### ISSUE THREAD

**Issue by clienthax** [SBPrime/AsyncWorldEdit#107]
Any chance of a version that works with just forge-worldedit for 1.8+ ?

**Comment by SBPrime** [comment]
Forge support is on demand. Provide a link to cauldron 1.8+ (cauldron = bukkit + forge) and I'll see what I can do about it.

**Comment by Xen1290** [comment]
Cauldron/KCauldron for 1.8 doesn't exist and I think it will be never exist. Because the ex manteiner of Cauldron is now one of lead developer with sk89q(WE,WG,CraftBook etc) and others of Sponge. KCauldron is a fork of Cauldron for 1.7.10 made by Protolik, it is and updated and fixed version. I think you will look to Sponge? If I remember correctly all plugins are moving to sponge, included WorldEdit.

**Comment by SBPrime** [comment]
If sponge API is as popular as Bukkit API I might consider moving to it. 
Current statistics:
dev.bukkit: ~15500 plugins
spigotmc: ~4200 + 800 premium bukkit  plugins and 454 bukkit bungee specific plugins, ~5500 in total
sponge: 25 releases + 38 requests.

So as you can see not all plugins are moving to the new api.

Regarding adding support for 1.8. AsyncWorldEdit requires Bukkit API to function correctly and the forge support is there only to make it work. Therefore if you have Forge API but don't have Bukit API it wont work. 

I can add a full native support for forge but only as a sponsored plugin.

**Comment by Xen1290** [comment]
Yeah, consider sponge is out from few months, it's in heavy development. But a lot of plugin announced the moving.

**Comment by clienthax** [comment]
@SBPrime What parts of the bukkit api does async rely on?, i know you cancel a few block change events etc.

**Comment by SBPrime** [comment]
According to the imports: PluginManager, Chunks, ChunkEvents, Player, Player join leave Events,  Block events, World, Scheduler, Bukkit Tasks, Configuration,  some bukkit specific WorldEdit classes (World, Player).

The new version will also use internal bukkit classes and internal server classes (Entties, NMS)

**Comment by clienthax** [comment]
@SBPrime Those are pretty easy to port over to forge,
Theres no plugin manager as such in forge, chunks and the chunk events are implemented in forge
so are the player join and leave, the scheduler +tasks can be replaced pretty easily, and with forge theres no need to use nms as you already are working at that level
, and worldedits forge version maps the forge player and world to its own version,

**Comment by SBPrime** [comment]
As stated earlier: I currently have no plans to port this to sponge I can consider porting this to new platform as a sponsored plugin.

Regarding the amount of work: your right its not much work but inner classes are important for the DirectChunk editing mode.

**Comment by Mhowser** [comment]
What is the current status of this as of now?

**Comment by SBPrime** [comment]
I'm considering providing a forge only version (actually a hybrid that would work on spigot and forge from the same jar). The core is already ported (mostly) to platform agnostic code. Unfortunately I do not have time to port it to forge.

**Comment by Mhowser** [comment]
Well it's good to know that it is still planned, thanks for the update!
